aten/src/ATen/native/TensorAdvancedIndexing.cpp

```diff
@@ -858,6 +858,33 @@ TORCH_IMPL_FUNC(index_add_cpu_out)
     if (numel == 0) {
       return;
     }
+    if (dim == (result.dim() - 1) && alpha.equal(1.0) &&
```

Comment:
Does `dim` has to be the last dim here or this fast path can also run faster with other dim values?

Reply:
When https://github.com/pytorch/pytorch/pull/82703 lands, using scatter_add will also get obvious speedup for the case dim == 0. Will add the cases dim == 0 after https://github.com/pytorch/pytorch/pull/82703 lands.